backends/edxorg.py

```diff
@@ -0,0 +1,55 @@
+"""
+EdX.org backend for Python Social Auth
+"""
+from urllib.parse import urljoin
+
+from django.conf import settings
+from social.backends.oauth import BaseOAuth2
+
+EDXORG_BASE_URL = settings.EDXORG_BASE_URL or 'https://courses.edx.org/'
+
+
+class EdxOrgOAuth2(BaseOAuth2):
+    """
+    EDX.org OAuth2 authentication backend
+    """
+    name = 'edxorg'
+    ID_KEY = 'edx_id'
+    REQUEST_TOKEN_URL = None
+    AUTHORIZATION_URL = urljoin(EDXORG_BASE_URL, '/oauth2/authorize/')
+    ACCESS_TOKEN_URL = urljoin(EDXORG_BASE_URL, '/oauth2/access_token/')
+    ACCESS_TOKEN_METHOD = 'POST'
+    REDIRECT_STATE = False
+    DEFAULT_SCOPE = ['email']
+
+    def user_data(self, access_token, *args, **kwargs):
+        """
+        Loads user data from service.
+
+        This is the function that has to pull the data from edx
+        """
```

Comment:
I was basing this on:

"this has to pull data from edx"
and the function below "given the data from edx, formats it into a dictionary"

Also the function below takes a "response", so I assumed http response.